src/Components/Endpoints/src/PublicAPI.Unshipped.txt

```diff
@@ -1,4 +1,8 @@
 #nullable enable
 Microsoft.AspNetCore.Components.Routing.RazorComponentsEndpointHttpContextExtensions
+Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider
+Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider.ServerAuthenticationStateProvider() -> void
+Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider.SetAuthenticationState(System.Threading.Tasks.Task<Microsoft.AspNetCore.Components.Authorization.AuthenticationState!>! authenticationStateTask) -> void
+override Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider.GetAuthenticationStateAsync() -> System.Threading.Tasks.Task<Microsoft.AspNetCore.Components.Authorization.AuthenticationState!>!
```

Comment:
Is this public because of layering?

Reply:
Or because we want it to be extendable

Reply:
I mention this in the PR description:

> In order to increase the consistency of using `AddAuthenticationStateSerialization()` with and without server interactivity, `ServerAuthenticationStateProvider` was type forwarded to `Microsoft.AspNetCore.Components.Authorization` and registered by default by `AddRazorComponents`. This is also helpful for auth in completely non-interactive Blazor web projects.

This change isn't strictly necessary for this PR, but I do think it helps with the general usability `AddRazorComponents` without `AddInteractiveServerComponents`. I could have used an internal `AuthenticationStateProvider` that's functionally equivalent to the `ServerAuthenticationStateProvider` without the type forwarding to achieve the same thing, but I figured we might as well reduce duplication.